src/main/java/com/questhelper/steps/tools/QuestPerspective.java

```diff
@@ -333,4 +357,158 @@ private static void addToPoly(Client client, Polygon areaPoly, WorldPoint wp, in
 			}
 		}
 	}
+
+	/**
+	 * Compares a quest-defined {@link WorldPoint} with a runtime {@link LocalPoint}, normalizing
+	 * both into the top-level world space so they can be compared safely across WorldViews.
+	 *
+	 * @return {@code true} when both coordinates refer to the same tile in the main world.
+	 */
+	public static boolean matchesWorldPoint(Client client, WorldPoint definedWorldPoint, LocalPoint runtimeLocalPoint)
```

Comment:
Should this be a `DefinedPoint definedPoint`? Seems like it and the other overload only ever get called from within `DefinedPoint`.

I think that would make sense if the intent for `DefinedPoint` is to make misuse of coordinates more difficult